src/Symfony/Component/Serializer/Mapping/Factory/ClassMetadataFactory.php

```diff
@@ -44,6 +44,10 @@ public function __construct(LoaderInterface $loader, Cache $cache = null)
     {
         $this->loader = $loader;
         $this->cache = $cache;
+
+        if (null !== $cache) {
+            @trigger_error('Passing a Doctrine Cache instance as parameter is deprecated and will be removed in Symfony 4.0. Use the Psr6MetadataFactory class instead.', E_USER_DEPRECATED);
```

Comment:
No I think that you should specify that the third argument of ``ClassMetadataFactory`` constructor is deprecated (I mean adding the FQN of ClassMetadataFactory)